modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C

```diff
@@ -291,6 +321,10 @@ INSFVRhieChowInterpolator::execute()
   if (_a_data_provided)
     return;
 
+  // If advecting with auxiliary variables, no need to try to run those kernels
+  if (_sys.number() != _u->sys().number())
+    return;
```

Comment:
I would prefer returning based off whether the `_velocity_interp_method` is average and then just combine with an `||` in the early return above. Then we are not unnecessarily gathering the RC data even when the advecting variables are in the nonlinear system

Reply:
Wait never mind ... I understand why you did this ... I actually compute the residuals within `gatherRCData` for some classes